File: Tarea5_BubbleSort_19-09-25/LEM_BubbleSort.cpp
#include <iostream>
#include <cstdlib>  
#include <ctime>    

using namespace std;

void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        bool swap = false;
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swap = true;
            }
        }
        if (!swap) {
            break; 
        }
    }
}

int main() {
    const int SIZE = 10;
    int arr[SIZE];

    srand(time(0)); 
    for (int i = 0; i < SIZE; i++) {
        arr[i] = rand() % 100 + 1;
    }

    cout << "Arreglo original: ";
    for (int i = 0; i < SIZE; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    bubbleSort(arr, SIZE);

    cout << "Arreglo ordenado: ";
    for (int i = 0; i < SIZE; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: Tarea5_BubbleSort_19-09-25/LEM_BubbleSort.java
import java.util.Random;

public class Main {
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            boolean swap = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swap = true;
                }
            }
            if (!swap) {
                break;
            }
        }
    }

    public static void main(String[] args) {
        final int SIZE = 10;
        int[] arr = new int[SIZE];
        Random rand = new Random();

        for (int i = 0; i < SIZE; i++) {
            arr[i] = rand.nextInt(100) + 1;
        }

        System.out.print("Arreglo original: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        bubbleSort(arr);

        System.out.print("Arreglo ordenado: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
